### PR TITLE
Implemented remembering last search term on the search transactions s…

### LINKED ISSUE #2828: Retain search query after opening transaction
### Please confirm the following:

- [X] I've checked [the current issues](https://github.com/Ivy-Apps/ivy-wallet/issues) for duplicate issues.
- [X] I've requested a single (only one) feature/change in this issue. It complies with the One Request Per GitHub Issue (ORPGI) rule.
- [X] My issue is well-defined and describes how it should be implemented from UI/UX perspective.

### What do you want to be added or improved?

Currently if I search transactions, click into one of them and go back to the search page, the search will be reset. It would be more user friendly if the search query are retained to enable better bulk editing or other functions.

### Why do you need it?

- Ease the process of going to specific transactions with keywords and editing them
- Much more intuitive and user friendly as the user should be the one that resets the query after searching

### How do you imagine it?

_No response_

### PR DESCRIPTION
**Pull request by Angeloem**
…creen

## Pull Request (PR) Checklist
Please check if your pull request fulfills the following requirements:
- [ ] The PR is submitted to the `main` branch.
- [ ] I've read the [Contribution Guidelines](https://github.com/Ivy-Apps/ivy-wallet/blob/main/CONTRIBUTING.md) and my PR doesn't break the "Contributing Rules".
- [ ] I've read the [Architecture Guidelines](https://github.com/Ivy-Apps/ivy-wallet/blob/main/docs/Architecture.md).
- [ ] I confirm that I've run the code locally and everything works as expected.
- [x] 🎬 I've attached a **screen recoding** of the changes. 

> Tip: drag & drop the video to the PR description.

## What's changed?

Describe with a few bullets **what's new:**
- Ivy wallet now remembers the search term after searching and viewing the transaction searched

> 💡 Tip: Please, attach screenshots and screen recordings. It helps a lot!

## Risk Factors

**What may go wrong if we merge your PR?**

- It is a very minor change, tests done work ok

**In what cases your code won't work?**

- I havent seen any so far

## Does this PR closes any GitHub Issues?

Check **[Ivy Wallet Issues](https://github.com/Ivy-Apps/ivy-wallet/issues)**.

- Closes #2828

https://github.com/Ivy-Apps/ivy-wallet/assets/40020525/717381ed-a243-4cfb-a723-2b084dc488cf

> Replace `ISSUE_NUMBER` with your issue number (for example Closes #1234). If you've done that correctly, you'll see the issue title linked when previewing your PR description.

## Troubleshooting CI failures

If you see any of the PR checks failing (❌) go to [Actions](https://github.com/Ivy-Apps/ivy-wallet/actions) and find it there. Or simply click "Details" next to the failed check and explore the logs to see why it has failed.

### Detekt
[Detekt](https://detekt.dev/) is a static code analyzer for Kotlin that we use to enforce code readibility and good practices.

**To run Detekt locally:**
```
./gradlew detekt
```

If the Detekt errors are caused by a legacy code, you can suppress them using a basline.

**Detekt baseline** (not recommended)
```
./scripts/detektBaseline.sh
```

### Lint

We use the [standard Android Lint](https://developer.android.com/studio/write/lint) plus [Slack's compose-lints](https://slackhq.github.io/compose-lints/) as an addition to enforce proper Compose usage.

**To run Lint locally:**
```
./scripts/lint.sh
```

If the Lint errors are caused by a legacy code, you can suppress them using a basline.

**Lint baseline** (not recommended)
```
./scripts/lintBaseline.sh
```

### Unit tests

If this job is failing this means that your changes break an existing unit test. You must identify the failing tests and fix your code.

**To run the Unit tests locally:**
```
./gradlew testDebugUnitTest
```

### CODE REVIEW

**Review by thesurajkamble — APPROVED** [review]
Looks good to me

File: screen-search/src/main/java/com/ivy/search/SearchViewModel.kt [hunk 1]
```diff
@@ -39,15 +41,18 @@ class SearchViewModel @Inject constructor(
 
     @Composable
     override fun uiState(): SearchState {
+        val prevSearchString = retrieveSearchHistory()
```
Comment (ILIYANGERMANOV) [inline screen-search/src/main/java/com/ivy/search/SearchViewModel.kt:44]: Shared preferences call on the UI thread in Composable which can be recomposed many times may lead to bad performance. Can we do that in the LaunchedEffect?

File: screen-search/src/main/java/com/ivy/search/SearchViewModel.kt [hunk 2]
```diff
@@ -60,6 +65,9 @@ class SearchViewModel @Inject constructor(
     private fun search(query: String) {
         val normalizedQuery = query.lowercase().trim()
 
+
+        sharedPrefs.putString(SharedPrefs.SEARCH_SCREEN_REMEMBER, normalizedQuery);
```
Comment (ILIYANGERMANOV) [inline screen-search/src/main/java/com/ivy/search/SearchViewModel.kt:69]: Can we use the Android DataStore instead? Ivy Wallet is migrating to it and also it's main-safe

Comment (ILIYANGERMANOV) [reply screen-search/src/main/java/com/ivy/search/SearchViewModel.kt:69]: We should also persist the search query in a separate coroutine so it doesn't block the search refresh logic

**Review by ILIYANGERMANOV — COMMENTED** [review]
Hey a few comments:
1. Use DataStore
2. Ensure the UI thread doesn't get blocked
3. Ensure the search logic performance

After these and when all CI checks pass, we can merge ✅

File: screen-search/src/main/java/com/ivy/search/SearchViewModel.kt (hunk 1 above)
Comment (Angeloem) [reply screen-search/src/main/java/com/ivy/search/SearchViewModel.kt:44]: Anhaa, will do that.

File: screen-search/src/main/java/com/ivy/search/SearchViewModel.kt (hunk 2 above)
Comment (Angeloem) [reply screen-search/src/main/java/com/ivy/search/SearchViewModel.kt:69]: Let me work on these over the weekend

**Pull request closed without merging**